src/runner.rs

```diff
@@ -166,53 +167,63 @@ enum RunResult {
 /// Function interpreter.
 pub struct Interpreter {
 	value_stack: ValueStack,
-	call_stack: Vec<FunctionContext>,
-	return_type: Option<ValueType>,
+	call_stack: StackWithLimit<FunctionContext>,
 	state: InterpreterState,
 }
 
 impl Interpreter {
-	pub fn new(func: &FuncRef, args: &[RuntimeValue]) -> Result<Interpreter, Trap> {
-		let mut value_stack = ValueStack::with_limit(DEFAULT_VALUE_STACK_LIMIT);
-		for &arg in args {
-			let arg = arg.into();
-			value_stack.push(arg).map_err(
-				// There is not enough space for pushing initial arguments.
-				// Weird, but bail out anyway.
-				|_| Trap::from(TrapKind::StackOverflow),
-			)?;
-		}
-
-		let mut call_stack = Vec::new();
-		let initial_frame = FunctionContext::new(func.clone());
-		call_stack.push(initial_frame);
-
-		let return_type = func.signature().return_type();
-
-		Ok(Interpreter {
-			value_stack,
+	/// Initialize an interpreter that will use `value_stack` and `call_stack`.
+	///
+	/// `value_stack` `call_stack` determine the allowed stack size during later executions.
+	pub fn new(
```

Comment:
Can we rename this `with_stacks` and have `new` create an `Interpreter` with sensible defaults (like your `fn new_interpreter` above).

Reply:
done